source/common/tcp_proxy/upstream.cc

```diff
@@ -404,7 +404,7 @@ CombinedUpstream::CombinedUpstream(HttpConnPool& http_conn_pool,
     : config_(config), downstream_info_(downstream_info), parent_(http_conn_pool),
       decoder_filter_callbacks_(decoder_callbacks), response_decoder_(*this),
       upstream_callbacks_(callbacks) {
-  auto is_ssl = downstream_info_.downstreamAddressProvider().sslConnection();
+  auto is_ssl = downstream_info_.upstreamInfo()->upstreamSslConnection();
```

Comment:
I think we should not make changes at other places except here, because those are not related to sending scheme header.
Also I think at the time of `CombinedUpstream` init downstream_info_.upstreamInfo()->upstreamSslConnection() would be blank.

Reply:
Hey! first of all thanks for reaching out so quickly!

From a check I did manually - this fix worked (This issue was triggered since in our setup we had a non tls downstream and a tls upstream).
Maybe there are other use-cases where downstream_info_.upstreamInfo() is not yet initialized but I couldnt find a scenario like this (and this was a concern I checked) - can you think of a specific scenario like that?

Regarding this change in other places - wouldnt that send wrong information in other places? It wont affect the headers but still it sends the wrong `ConnectionInfo` - maybe other bugs are hidden in this use-case beside the scheme headers so not entirely sure why we want to ignore other places?